aten/src/ATen/native/NamedTensor.cpp

```diff
@@ -142,12 +142,36 @@ static Tensor align(const Tensor& tensor, DimnameList names, bool is_aligning_tw
 }
 
 Tensor align_to(const Tensor& tensor, DimnameList names) {
-  TORCH_CHECK(
-      names.size() >= tensor.dim(),
-      "Cannot align tensor with dims ",
-      tensor.names(),
-      " to a shorter list of dims ", names, ".");
-  return align(tensor, names, /*aligning_two_tensors=*/false);
+  auto tensor_names = tensor.names();
+  auto tensor_sizes = tensor.sizes();
+  auto tensor_strides = tensor.strides();
+  std::vector<int64_t> new_sizes(names.size(), 1);
+  std::vector<int64_t> new_strides(names.size(), 0);
+
+  for (auto idx = 0; idx < tensor_names.size(); ++idx) {
+    const auto& dim = tensor_names[idx];
+    TORCH_CHECK(dim.is_normal(),
+        "align_to: All input dims must be named. Found unnamed dim at index ",
+        dim, " of Tensor", tensor_names);
```

Comment:
I think you probably want a space after "Tensor"?

Reply:
Nope, no space necessary. It should look like 

```
of Tensor['N', 'C', 'H', 'W']
```

for example. I'll try to toss this into a helper function in a followup PR because I've noticed that I've done this a lot in the codebase.

Reply:
ah right, I think I've been confused by this before too but makes sense.

Reply:
Yeah, I'll toss it onto my clean-up list.